LinearRegression/LinearRegressionBackend/NeuralNetworkPlayground/Network.cs

```diff
@@ -0,0 +1,59 @@
+﻿using System.Collections.Generic;
+
+namespace LinearRegressionBackend.NeuralNetworkPlayground
+{
+    public class Network
+    {
+
+        public List<Layer> Layers;
+
+        public Network(List<Layer> layers)
+        {
+            Layers = layers;
+        }
+
+        public Propagation Propagate(double input)
+        {
+            Propagation result = new();
+
+            result.WeightedSums.Add(0);
+            result.Activations.Add(input);
+
+            foreach (Layer layer in Layers)
+            {
+                (double sum, double activation) = layer.Propagate(input);
+                result.WeightedSums.Add(sum);
+                result.Activations.Add(activation);
+                input = activation;
+            }
+
+            return result;
+        }
+
+        public void Backpropagate(
+            Propagation prop,
+            double expected,
+            double learningRate)
+        {
+            double delta = prop.Output() - expected;
```

Comment:
I would calculate delta separately from last layer (output layer). delta=(a_L-y) * dsigmoid(z_L), where L is the last layer.
Then I will iterate from the last hidden layer (l=L-1). delta_l=delta_l+1 * w_l+1 * dsigmoid(z_l)
After it is done we can calculate the gradients: dw_l=delta_l * a_l-1, db_l=delta_l